metadata-ingestion/src/datahub/ingestion/source/superset.py

```diff
@@ -308,6 +308,11 @@ def construct_chart_from_chart_data(self, chart_data):
         group_bys = params.get("groupby", []) or []
         if isinstance(group_bys, str):
             group_bys = [group_bys]
+        # handling groupby custom label case
+        # a dict containing two keys: sqlExpression and label
+        elif group_bys and isinstance(group_bys[0], dict):
```

Comment:
What if group_bys is empty? 

Should we handle all objects in the list, if non-empty?

Reply:
Hi jjoyce0510,

In case the group_bys is empty (an empty list to be precise), I'm falling back to the original code, which takes the list and turns it into an empty string, as in <code>", ".join(group_bys)</code>

In case the group_bys is not empty, there will be three *or four* cases so far:
+ group_bys is a **List[str]** > This case is already handled in the original code
+ group_bys is a **List[Union[str, dict]]** > Not yet handled
+ group_bys is a **List[dict]** > Handled in this PR

Regarding the fourth case, I'm not really sure if it exists in previous versions of Superset or not, according to the original ingestion code, referencing below.
+ A string or an empty string, converted to a list before adding to the <code>custom_properties</code> variable.
```
if isinstance(group_bys, str):
    group_bys = [group_bys]

custom_properties = {
    "Dimensions": ", ".join(group_bys)
}
```

I'm thinking of something like a loop, which checks if the second case is true. If true, it's only a matter of extracting all the <code>label</code> values inside the dict as a string, appending it to a new <code>group_bys</code> if needed, then adding all the remaining string values if applied, which then it's back to the first case, a **List[str]**.

Thank you for the reply, and the extremely helpful question, as I've forgotten to consider the second case :)

Reply:
Basically I'm most concerned that `isinstance(group_bys[0], dict)` will throw an error since we are trying to access an index in the list without checking the list length..

I think it would be ideal to cover this case:

`group_bys is a List[Union[str, dict]] > Not yet handled`

by looping over the list, calling a function for each item, and finally appending to our own `group_bys` array~

John